In [ ]:
!pip install wooldridge

     |████████████████████████████████| 5.1 MB 12.9 MB/s 


In [ ]:
import wooldridge as wd

In [ ]:
wage=wd.data("wage1")

In [ ]:
wage

wage  educ  exper  tenure  nonwhite  female  married  numdep  smsa  \
0     3.10    11      2       0         0       1        0       2     1   
1     3.24    12     22       2         0       1        1       3     1   
2     3.00    11      2       0         0       0        0       2     0   
3     6.00     8     44      28         0       0        1       0     1   
4     5.30    12      7       2         0       0        1       1     0   
..     ...   ...    ...     ...       ...     ...      ...     ...   ...   
521  15.00    16     14       2         0       1        1       2     0   
522   2.27    10      2       0         0       1        0       3     0   
523   4.67    15     13      18         0       0        1       3     0   
524  11.56    16      5       1         0       0        1       0     0   
525   3.50    14      5       4         1       1        0       2     0   

     northcen  ...  trcommpu  trade  services  profserv  profocc  clerocc  \
0           0  ...         0      0         0         0        0        0   
1           0  ...         0      0         1         0        0        0   
2           0  ...         0      1         0         0        0        0   
3           0  ...         0      0         0         0        0        1   
4           0  ...         0      0         0         0        0        0   
..        ...  ...       ...    ...       ...       ...      ...      ...   
521         0  ...         0      0         0         1        1        0   
522         0  ...         0      1         0         0        1        0   
523         0  ...         0      0         0         0        1        0   
524         0  ...         0      0         0         0        0        0   
525         0  ...         0      0         0         1        0        1   

     servocc     lwage  expersq  tenursq  
0          0  1.131402        4        0  
1          1  1.175573      484        4  
2          0  1.098612        4        0  
3          0  1.791759     1936      784  
4          0  1.667707       49        4  
..       ...       ...      ...      ...  
521        0  2.708050      196        4  
522        0  0.819780        4        0  
523        0  1.541159      169      324  
524        0  2.447551       25        1  
525        0  1.252763       25       16  

[526 rows x 24 columns]

In [ ]:
wage=wage[["wage","educ"]]

In [ ]:
wage

wage  educ
0     3.10    11
1     3.24    12
2     3.00    11
3     6.00     8
4     5.30    12
..     ...   ...
521  15.00    16
522   2.27    10
523   4.67    15
524  11.56    16
525   3.50    14

[526 rows x 2 columns]

In [ ]:
wage.columns

Index(['wage', 'educ'], dtype='object')

In [ ]:
wage.columns=['wagex', 'educy']
wage.columns

Index(['wagex', 'educy'], dtype='object')

In [ ]:
wage

wagex  educy
0     3.10     11
1     3.24     12
2     3.00     11
3     6.00      8
4     5.30     12
..     ...    ...
521  15.00     16
522   2.27     10
523   4.67     15
524  11.56     16
525   3.50     14

[526 rows x 2 columns]

In [ ]:
wage_=wage ["wagex"]
educ_=wage["educy"]

In [ ]:
class econometria:
  def __init__(self,z):
     self.z = z
  def media(self):
    return sum(self.z)/len(self.z)
  def mediana(self):
    self.z.sort()
    num=len(self.z)
    if num %2==0:
      print((self.z[int(num/2)]+self.z[int(num/2)-1])/2)
    else:
      print(self.z[int(num/2-0.5)])
  def varianza (self):
    s = 0
    m = self.media()
    for elemento in z:
      s += (elemento - m) ** 2
    return s / float(len(self.z))
   
  def desviacion_estandar(self):
    return (self.varianza())**(1/2)
  def coeficiente_variacion(self):
    return self.desviacion_estandar()/self.media()


In [ ]:
est=econometria(wage_.to_list())

In [ ]:
est.media()

5.896102674787035

In [ ]:
est.mediana()

4.650000095367432


In [ ]:
est.varianza()

13.612954925821066

In [ ]:
est.desviacion_estandar()

3.6895738135753655

In [ ]:
est.coeficiente_variacion()

0.6257648513063981

In [ ]:
est2=econometria(educ_.to_list())

In [ ]:
est2.media()

12.562737642585551

In [ ]:
est2.mediana()

12.0


In [ ]:
est2.varianza()

58.05697671969503

In [ ]:
est2.desviacion_estandar()

7.619512892547333

In [ ]:
est2.coeficiente_variacion()

0.6065169160835195

#interpretacion wagex
Según la media la mitad de los valores es menor que o igual a 
5.896102674787035 y la otra mitad de los valores es mayor que o igual a 
4.650000095367432, observamos que no se cumple una relación simétrica, si no asimétrica ya que los valores no son similares, partiendo de la varianza se observa que mientras la varianza sea mayor, mayor sera la dispersion de los datos, analisando que la dispersion es de 
13.612954925821066, con el coeficiente de variacion vemos que la dispersion relativa del conjunto de datos es de 0.6257648513063981, y para finalizar observamos que la dispersion de los datos es de 
3.6895738135753655

#interpretacion educy
Según la media la mitad de los valores es menor que o igual a 12.562737642585551 y la otra mitad de los valores es mayor que o igual a12.0, observamos que se cumple una relación simétrica, si seria asimétrica ya que los valores son relativamente similares, partiendo de la varianza se observa que mientras la varianza sea mayor, mayor sera la dispersion de los datos,viendo una gran dispercion, analisando que la dispersion es de 58.05697671969503, con el coeficiente de variacion vemos que la dispersion relativa del conjunto de datos es de 0.6065169160835195, y para finalizar observamos que la dispersion de los datos es de 7.619512892547333